src/test/java/io/camunda/testing/assertions/IncidentAssertTest.java

```diff
@@ -0,0 +1,449 @@
+      final ActivatedJob job = jobActivationResponse.getJobs().get(0);
+      client.newCompleteCommand(job.getKey()).send().join();
+
+      // TODO remove this
+      Thread.sleep(100);
+
+      final IncidentAssert incidentAssert = assertThat(instanceEvent).extractLatestIncident();
+
+      // then
+      incidentAssert.occurredOnElement(ProcessPackLoopingServiceTask.GATEWAY_ELEMENT_ID);
+    }
+
+    @Test
+    void testOccurredDuringJob() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob job = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(job.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(job).extractLatestIncident();
+
+      // then
+      incidentAssert.occurredDuringJob(job);
+      incidentAssert.occurredDuringJob(job.getKey());
+    }
+
+    @Test
+    void testIsResolved() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      final long incidentKey = incidentAssert.getIncidentKey();
+      client.newResolveIncidentCommand(incidentKey).send().join();
+
+      // TODO remove
+      Thread.sleep(100);
+
+      // then
+      incidentAssert.isResolved();
+    }
+
+    @Test
+    void testIsUnresolved() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      // then
+      incidentAssert.isUnresolved();
+    }
+  }
+
+  private ActivateJobsResponse activateSingleJob() {
```

Comment:
❌  This could be moved to the testing Utilities. The same method is also used in `JobAssertTest` so it'd be nice to move this to a place where it can be used by both tests classes.